### PR TITLE
Automatically build and publish releases

### PR DESCRIPTION
**Pull request by Nuru**
## what
Use [goreleaser](https://goreleaser.com) and GitHub actions to automatically publish `terraform-provider-nexus` binaries and change logs on every release. 

(Also fixes typo in README.md)

## why
Make the provider easily available for download as a Terraform plugin

## how
Every time a release is tagged, GitHub actions will be invoked to automatically build and publish the artifacts. 

A user can then install the plugin with something like
```
cd ~/.terraform.d/plugins/linux_amd64
curl -sSL https://github.com/datadrivers/terraform-provider-nexus/releases/download/v1.2.1/terraform-provider-nexus_v1.2.1_linux_amd64.tar.gz | tar xzv
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Unshallow
+        run: git fetch --prune --unshallow
+      -
+        name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13.x
```
Comment (akaRem) [inline .github/workflows/release.yml:24]: The project uses go v1.14

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+build:
+  binary: terraform-provider-nexus_v{{ .Version }}
+  goos:
+  - darwin
+  - linux
+  - windows
```
Comment (akaRem) [inline .goreleaser.yml:6]: Is windows supported by makefile? (see https://github.com/datadrivers/terraform-provider-nexus/blob/master/GNUmakefile#L3)

File: .github/workflows/release.yml (hunk 1 above)
Comment (Nuru) [reply .github/workflows/release.yml]: Fixed

File: .goreleaser.yml (hunk 2 above)
Comment (Nuru) [reply .goreleaser.yml:6]: `goreleaser` does not use the GNUmakefile. It builds using standard `go` build commands.

**Comment by Nuru** [comment]
> So this will create a new tarball (zip arch) every time a new plugin is pushed.
> Am I correct?
> 
> Does it require any additional configuration?

Additional configuration is available (see docs for [GitHub action](https://github.com/goreleaser/goreleaser-action) and [tool](https://goreleaser.com/)), but not required. You can see the results at https://github.com/cloudposse/terraform-provider-nexus/releases/tag/v0.1.2 where all I did to create that release is create the version tag.

**Comment by osterman** [comment]
@Nosmoht just wanted to get this on your radar! We'd like to add this provider to our docker image.

**Pull request merged**